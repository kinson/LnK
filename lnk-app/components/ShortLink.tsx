import React from 'react';

interface Props {
    content: string | null;
    hidden: boolean;
}

export default function ShortLink({ content, hidden }: Props) {

    const baseStyle = "text-violet text-md relative bg-white rounded border-0 shadow outline-none focus:outline-none w-full transition-opacity duration-300";

    const style = hidden ? `${baseStyle} h-0 p-0 opacity-0`: `${baseStyle} p-3 opacity-1`;

    return (
        <p className={style}>
            {content}
        </p>
    )
}