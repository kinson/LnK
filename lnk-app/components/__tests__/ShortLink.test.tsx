import React from 'react';
import renderer from 'react-test-renderer';
import ShortLink from '../ShortLink';

describe('ShortLink component tests', () => {
  it('should render the ShortLink component when it is not hidden', () => {
    const component = renderer.create(
      <ShortLink link="https://google.com" hidden={false} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should hide the ShortLink component when it is hidden', () => {
    const component = renderer.create(<ShortLink link={null} hidden />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
