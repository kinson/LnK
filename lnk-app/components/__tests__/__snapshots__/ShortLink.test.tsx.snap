// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortLink component tests should hide the ShortLink component when it is hidden 1`] = `
<a
  className="text-violet dark:text-lavender text-md relative bg-white dark:bg-gray-600 xrounded border-0 shadow outline-none focus:outline-none w-full transition-opacity duration-300 h-0 p-0 opacity-0"
  href="#"
/>
`;

exports[`ShortLink component tests should render the ShortLink component when it is not hidden 1`] = `
<a
  className="text-violet dark:text-lavender text-md relative bg-white dark:bg-gray-600 xrounded border-0 shadow outline-none focus:outline-none w-full transition-opacity duration-300 p-3 opacity-1"
  href="https://google.com"
>
  https://google.com
</a>
`;
